src/Glpi/Form/QuestionType/QuestionTypesManager.php

```diff
@@ -159,4 +159,104 @@ protected function loadQuestionsTypes(): void
             fn(QuestionTypeInterface $a, QuestionTypeInterface $b) => $a->getWeight() <=> $b->getWeight()
         );
     }
+
+    public function getTemplateSelectionForCategories(): string
+    {
+        $icons = array_combine(
+            array_map(
+                fn(QuestionTypeCategory $type) => $type->value,
+                (array) $this->getCategories()
+            ),
+            array_map(
+                fn(QuestionTypeCategory $type) => $type->getIcon(),
+                (array) $this->getCategories()
+            )
+        );
+        $js_icons = json_encode($icons);
+
+        return <<<JS
+            function(item) {
+                const icons = {$js_icons};
+                return $('<span class="d-flex flex-row-reverse align-items-center gap-2">'
+                    + '<i class="' + icons[item.id] + '"></i>'
+                    + item.text
+                    + '</span>');
+            }
+        JS;
```

Comment:
`gettext` issue related to indented heredoc ending marker is not yet validated :/
```suggestion
JS;
```